_projects/beb_vectortile_refactor.md

```diff
@@ -0,0 +1,74 @@
+
+# project attributes
+requirements: # bullet list of requirements – one requirement per line – follow below format
+  - "Web development experience in JavaScript"
+  - "Scripting language experience"
+  - "Familiarity with angularjs or Angular preferred"
+
+tags: # one tag per line – spaces are allowed in tags – use tags other posts use
+  - "javascript"
+  - "angularjs"
+  - "vector tiles"
+
+difficulty: "Easy" # Easy, Medium, Hard – pick one
+
+mentors: # github username without the @ – example: designmatty
+  - "cloudniner"
+  - "klaash"
+  - "ddohler"
+
+# This file uses Kramdown. See https://kramdown.gettalong.org/syntax.html for syntax
+---
+
+At Azavea we're always looking for ways to provide cutting edge, performant web visualizations for geospatial data. The goal of this project is to determine the level of effort required to replace an existing Carto backend with vector tiles for the Philly [Energy Building Benchmarking](http://visualization.phillybuildingbenchmarking.com) application that we developed a few years ago in concert with the Mayor's Office of Sustainability.
+
+In general, vector tiles appear to be an excellent fit for this application because they:
+
+- Are cost-effective for data that updates infrequently
+- Do not require a server and can be hosted as static files
+- Are performant for datasets with a large number of features
+- Provide flexible styling options based on zoom level, geometry, feature properties and more
+
+# Milestones
+
+## Preparation
+
+In the beginning of the project you will prepare by:
+
+- Familiarizing yourself with existing vector tile display, query and style methods
+- Familiarizing yourself with with angularjs and Mapbox GL JS
+- Work with your mentor to develop a project roadmap
+
+## Project Phase 1
+
+The primary goal of this phase is to replace Carto map with Mapbox GL JS and display the original dataset on the map.
+
+This will require you to:
+
+- Generate a static vector tile set for the original source dataset
+- Generate a static vector tile set for the basemap
+- Replace backend calls via the Carto SQL library with equivalent calls in Mapbox GL JS
+- Re-implement map filters using vector tile API
+
+## Project Phase 2
+
+The primary goal of this phase is to restore the original charts, legends and map styles using the new vector tiles.
+
+- Develop legend components to replace those of the Carto API
+- Calculate statistics for metrics that cannot be computed via vector tiles for application charts
+- Create vector tile styles for each of the application filter and category types
+
+## Project Phase 3
+
+The final phase of the project is to write a short technical document that reports:
+
+- The pros and cons of the Carto and vector tile implementations
+- Pain points experienced during the refactor
+- Recommendation on whether to use vector tiles for similarly scoped projects in the future
```

Comment:
This overall looks great and is really well scoped. I think it is perhaps a bit ambitious for the amount of time in a fellowship (which is effectively 8-9 weeks after you take orientation and end-of-summer presentations into account). However, the nice thing about fellows is that they can work pretty close to flat-out because they generally don't have other responsibilities or meetings to get in the way. So I think my only comment is to make sure that we get a roadmap planned out very early on and that the mentors think of escape hatches ahead of time so that we can get what we want out of the fellowship if the timeline starts to slip.

Reply:
Great point. I left this as a note for myself. As far as escape hatches are concerned, the bulk of our interest in the work is in Phase 1. I'd go as far as to say that in the worst case scenario, most of Phase 2 could be dropped if a day or so was spent with the fellow scoping the implications and estimated effort of those tasks.